File: api/src/main/java/fr/huiitre/tools/modules/riot/sync/infrastructure/PostgresValorantSkinLevelSyncRepository.java
package fr.huiitre.tools.modules.riot.sync.infrastructure;

import org.springframework.jdbc.core.JdbcTemplate;

import fr.huiitre.tools.modules.riot.sync.application.ValorantSkinLevelSyncData;
import fr.huiitre.tools.modules.riot.sync.application.ValorantSkinLevelSyncRepository;

public class PostgresValorantSkinLevelSyncRepository implements ValorantSkinLevelSyncRepository {

    private final JdbcTemplate jdbcTemplate;

    public PostgresValorantSkinLevelSyncRepository(JdbcTemplate jdbcTemplate) {
        this.jdbcTemplate = jdbcTemplate;
    }

    @Override
    public void deleteAll() {
        jdbcTemplate.update("DELETE FROM tools_riot.valorant_skin_levels");
    }

    @Override
    public void save(Long skinId, ValorantSkinLevelSyncData data) {
        final String sql = """
                INSERT INTO tools_riot.valorant_skin_levels
                    (skin_id, asset_id, level_index, name, level_item, display_icon_url, streamed_video_url)
                VALUES (?, ?, ?, ?, ?, ?, ?)
                """;
        jdbcTemplate.update(sql,
                skinId,
                data.getAssetId(),
                data.getLevelIndex(),
                data.getName(),
                data.getLevelItem(),
                data.getDisplayIconUrl(),
                data.getStreamedVideoUrl());
    }
}

File: api/Modules/Riot/Sync/Infrastructure/PostgresValorantSkinLevelSyncRepository.cs
using Tools.Api.Modules.Riot.Sync.Application;
using Tools.Api.Modules.Riot.Common.Infrastructure;
using Tools.Api.Modules.Riot.Sync.Application.Ports;

namespace Tools.Api.Modules.Riot.Sync.Infrastructure;

public sealed class PostgresValorantSkinLevelSyncRepository(RiotDatabase database) : IValorantSkinLevelSyncRepository
{
    public Task DeleteAll() =>
        database.Execute("DELETE FROM tools_riot.valorant_skin_levels");

    public Task Save(long skinId, ValorantSkinLevelSyncData data) =>
        database.Execute(
            """
            INSERT INTO tools_riot.valorant_skin_levels
                (skin_id, asset_id, level_index, name, level_item, display_icon_url, streamed_video_url)
            VALUES (@SkinId, @AssetId, @LevelIndex, @Name, @LevelItem, @DisplayIconUrl, @StreamedVideoUrl)
            """,
            new { SkinId = skinId, data.AssetId, data.LevelIndex, data.Name, data.LevelItem, data.DisplayIconUrl, data.StreamedVideoUrl });
}
